js/modules/k6/experimental/fs/module.go

```diff
@@ -0,0 +1,326 @@
+
+	// file contains the actual implementation for the file system.
+	file
+
+	// vu holds a reference to the VU this file is associated with.
+	//
+	// We need this to be able to access the VU's runtime, and produce
+	// promises that are handled by the VU's runtime.
+	vu modules.VU
+
+	// registry holds a pointer to the file registry this file is associated
+	// with. That way we are able to close the file when it's not needed
+	// anymore.
+	registry *cache
+}
+
+// Stat returns a promise that will resolve to a [FileInfo] instance describing
+// the file.
+func (f *File) Stat() *goja.Promise {
+	promise, resolve, _ := promises.New(f.vu)
+
+	go func() {
+		resolve(f.file.stat())
+	}()
+
+	return promise
+}
+
+// Read the file's content, and writes it into the provided Uint8Array.
+//
+// Resolves to either the number of bytes read during the operation
+// or EOF (null) if there was nothing more to read.
+//
+// It is possible for a read to successfully return with 0 bytes.
+// This does not indicate EOF.
+func (f *File) Read(into goja.Value) *goja.Promise {
+	promise, resolve, reject := f.vu.Runtime().NewPromise()
+
+	if common.IsNullish(into) {
+		reject(newFsError(TypeError, "read() failed; reason: into argument cannot be null or undefined"))
+		return promise
+	}
+
+	// We expect the into argument to be a `Uint8Array` instance
+
+	// intoObj := into.ToObject(f.vu.Runtime())
+	// uint8ArrayConstructor := f.vu.Runtime().Get("Uint8Array")
+	// if isUint8Array := intoObj.Get("constructor").SameAs(uint8ArrayConstructor); !isUint8Array {
+	intoObj := into.ToObject(f.vu.Runtime())
+	if !isUint8Array(f.vu.Runtime(), intoObj) {
+		reject(newFsError(TypeError, "read() failed; reason: into argument must be a Uint8Array"))
+		return promise
+	}
+
+	// Obtain the underlying ArrayBuffer from the Uint8Array
+	ab, ok := intoObj.Get("buffer").Export().(goja.ArrayBuffer)
+	if !ok {
+		reject(newFsError(TypeError, "read() failed; reason: into argument must be a Uint8Array"))
+		return promise
+	}
+
+	// To avoid concurrency linked to modifying the runtime's `into` buffer from multiple
+	// goroutines we make sure to work on a separate copy, and will copy the bytes back
+	// into the runtime's `into` buffer once the promise is resolved.
+	intoBytes := ab.Bytes()
+	buffer := make([]byte, len(intoBytes))
+
+	// We register a callback to be executed by the VU's runtime.
+	// This ensures that the modification of the JS runtime's `into` buffer
+	// occurs on the main thread, during the promise's resolution.
+	callback := f.vu.RegisterCallback()
+	go func() {
+		n, readErr := f.file.Read(buffer)
+		callback(func() error {
+			_ = copy(intoBytes[0:n], buffer)
+
+			// Read was successful, resolve early with the number of
+			// bytes read.
+			if readErr == nil {
+				resolve(n)
+				return nil
+			}
+
+			// The [file.Read] method will return an EOFError as soon as it reached
+			// the end of the file.
+			//
+			// However, following deno's behavior, we express
```

Comment:
I have a hard time getting what is `following deno's behavior`. Is this is a relevant details here?